src/main/java/com/direwolf20/buildinggadgets/common/concurrent/ServerTickingScheduler.java

```diff
@@ -0,0 +1,47 @@
+package com.direwolf20.buildinggadgets.common.concurrent;
+
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.event.TickEvent.Phase;
+import net.minecraftforge.event.TickEvent.ServerTickEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+import java.util.EnumSet;
+import java.util.function.BooleanSupplier;
+
+public final class ServerTickingScheduler {
+    public static void runTicked(BooleanSupplier runUntilFalse) {
+        new ServerTickingScheduler(runUntilFalse, EnumSet.of(Phase.START));
+    }
+
+    public static void runTickedAtEnd(BooleanSupplier runUntilFalse) {
+        new ServerTickingScheduler(runUntilFalse, EnumSet.of(Phase.END));
+    }
+
+    public static void runTickedStartAndEnd(BooleanSupplier runUntilFalse) {
```

Comment:
Is a run on the tick start and end needed? Most mods I've seen will only ever put something on the start or end?

Reply:
This is used, if you want something to run as fast as possible during Server-Tick. For example if you come from another Thread, then you usually want it to run as fast as possible, not after the Server waits for the next tick to occur or after it ticked everything else.

Reply:
Alright.